src/idempotent.ts

```diff
@@ -1,34 +1,68 @@
-export interface IdempotentStore {
-  get<T = any>(request: string): Promise<{ found: boolean; result: T }>;
-  set<T>(request: string, response: T): Promise<boolean>;
-  lock(request: string): Promise<boolean>;
-  release(request: string): Promise<boolean>;
-}
+import { FibonacciStrategy, Backoff } from 'backoff';
 
-export class LockError extends Error {
-  constructor() {
-    super('Failed to lock the request id');
-  }
+export interface IdempotentStore {
+  get(id: string): Promise<any | null>;
+  set(
+    id: string,
+    params:
+      | {
+          status: 'STARTED';
```

Comment:
Why do we need to have this?,. We should try to gain lock, if we fail we do backoff,. no need to track the status,. I found it a lot more complicated than just try lock multiple times.

Reply:
This has the same effect as having locks, with the additional advantage of being less confusing. We could call this field `locked`, but then the context is missing. We can lock a request with an existing result (which is useless). The lock is only useful if the lock has been acquired but the result is not yet available.

Reply:
also, I thought this would make the `IdempotentStore` interface simpler. By making the implementor not think about locks.

Reply:
Yeah ok,. makes sense.

The `IdempotentStore` implementor needs to take care of the locks.